databases/ClickHouse.java

```diff
@@ -0,0 +1,79 @@
+import com.clickhouse.client.*;
+
+import java.util.Map;
+
+public class ClickHouse implements Database {
+
+    ClickHouseNode server;
+    @Override
+    public int setup(Dataset dataset) {
+        server = ClickHouseNode.builder()
+                .host(System.getProperty("chHost", "localhost"))
+                .port(ClickHouseProtocol.HTTP, Integer.getInteger("chPort", 8123))
+                .database("system").credentials(ClickHouseCredentials.fromUserAndPassword(
+                        System.getProperty("chUser", "benchmark"), System.getProperty("chPassword", "")))
+                .build();
+
+        try (ClickHouseClient client = ClickHouseClient.newInstance(server.getProtocol())) {
+            ClickHouseRequest<?> request = client.connect(server);
+            StringBuilder schema = new StringBuilder();
+            for (Map.Entry<String, ColumnType> column :dataset.getColumnNamesWithTypes()) {
+                schema.append(column.getKey() + " " + columnTypeToString(column.getValue()) + ", ");
+            }
+            schema.delete(schema.length() - 2, schema.length());
```

Comment:
instead of this loop and dropping few chars afterwards we can do it in one instruction which in my is more readable
```
var schema = String.join(", ",
                dataset.getColumnNamesWithTypes()
                        .stream()
                        .map(kv -> kv.getKey() + " " + columnTypeToString(column.getValue()))
                        .toArray(String[]::new)
        );
```

or smth similar, didn't compile it. Of course this way is correct as well and may stay this way, just wanted to show different approach to such chunks of code

Reply:
We dont need it as an array but as a comma separated string though. There might be a nicer way, but I would keep it for now, especially bc it isnt 100% clear yet whether I can use ClickHouse